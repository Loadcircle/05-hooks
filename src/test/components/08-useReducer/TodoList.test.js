import React from 'react';
const { shallow } = require("enzyme");
const { TodoList } = require("../../../components/08-useReducer/TodoList");
const { demoTodos } = require("../../fixtures/demoTodos");

describe('Pruebas en el TodoList', ()=>{

    const handleDelete = jest.fn();
    const handleToggle = jest.fn();

    const wrapper = shallow(<TodoList
        todos = {demoTodos} 
        handleDelete = {handleDelete} 
        handleToggle = {handleToggle}    
    />);

    test('Debe hacer match el snapshot', ()=>{

        expect(wrapper).toMatchSnapshot();

    });
    
    test('Debe tene 2 todolistitem', ()=>{

        expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length);
        expect(wrapper.find('TodoListItem').at(0).prop('handleDelete')).toEqual(expect.any(Function));

    });    

});