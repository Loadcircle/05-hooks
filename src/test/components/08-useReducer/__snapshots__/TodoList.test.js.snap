// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el TodoList Debe hacer match el snapshot 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "desc": "Aprende",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "desc": "Aprende2",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
